src/sdp/DefaultSDP.ts

```diff
@@ -267,6 +267,70 @@ export default class DefaultSDP implements SDP {
     return new DefaultSDP(originalSdp);
   }
 
+  getIdForAllocationRtpHeaderExtension(): number {
+    const srcLines: string[] = this.lines();
+    const headerExtensionIds: number[] = [];
+
+    for (const line of srcLines) {
+      if (/^a=extmap:/.test(line.trim())) {
+        const headerExtension = line.split('a=extmap:')[1].split(' ');
+        const id = +headerExtension[0];
+        headerExtensionIds.push(id);
+      }
+    }
+
+    headerExtensionIds.sort((a, b) => a - b);
+    let previousId = 0; // header extension cannot be 0, refer https://datatracker.ietf.org/doc/html/rfc5285
+    for (const id of headerExtensionIds) {
+      if (id - previousId > 1) {
+        return previousId + 1;
+      }
+      previousId = id;
+    }
+
+    // One-Byte Header header extension cannot be bigger than 14, refer https://datatracker.ietf.org/doc/html/rfc5285
+    return previousId === 14 ? -1 : previousId + 1;
+  }
+
+  // this will not add the packet overhead unless negotiated to avoid waste
+  // it is tested against local Tincan
+  withVideoLayersAllocationRtpHeaderExtension(): DefaultSDP {
+    const sections = DefaultSDP.splitSections(this.sdp);
+    if (sections.length < 2) {
+      return new DefaultSDP(this.sdp);
+    }
+
+    const newSections = [];
+    for (let section of sections) {
+      if (/^m=video/.test(section)) {
+        const id = this.getIdForAllocationRtpHeaderExtension();
```

Comment:
why do we need to run this for every section? or should we just make this function take the section.